{{cookiecutter.profile_name}}/lsf_submit.py

```diff
@@ -0,0 +1,187 @@
+"""
+
+import sys
+import subprocess
+from pathlib import Path
+from snakemake.utils import read_job_properties
+from typing import List
+import re
+if not __name__.startswith("tests.src."):
+    sys.path.append(str(Path(__file__).parent.absolute()))
+    from OSLayer import OSLayer
+    from CookieCutter import CookieCutter
+else:
+    from .OSLayer import OSLayer
+    from .CookieCutter import CookieCutter
+
+
+class LSF_Submit:
+    def __init__(self, argv: List[str]):
+        self._jobscript = argv[-1]
+        self._cluster_cmd = " ".join(argv[1:-1])
+        self._job_properties = read_job_properties(self._jobscript)
+        self.random_string = OSLayer.get_random_alphanumerical_string()
+
+    @property
+    def jobscript(self):
+        return self._jobscript
+
+    @property
+    def job_properties(self):
+        return self._job_properties
+
+    @property
+    def cluster(self):
+        return self.job_properties.get("cluster", dict())
+
+    @property
+    def threads(self):
+        return self.job_properties.get("threads", CookieCutter.get_default_threads())
+
+    @property
+    def resources(self):
+        return self.job_properties.get("resources", dict())
+
+    @property
+    def mem_mb(self):
+        return self.resources.get("mem_mb", self.cluster.get("mem_mb", CookieCutter.get_default_mem_mb()))
+
+    @property
+    def resources_cmd(self) -> str:
+        return (
+            "-M {mem_mb} -n {threads} "
+            "-R 'select[mem>{mem_mb}] rusage[mem={mem_mb}] span[hosts=1]'"
+        ).format(mem_mb=self.mem_mb, threads=self.threads)
+
+    @property
+    def wildcards(self):
+        return self.job_properties.get("wildcards", dict())
+
+    @property
+    def wildcards_str(self):
+        return ".".join("{}={}".format(k, v) for k, v in self.wildcards.items()) or "unique"
+
+    @property
+    def rule_name(self):
+        return self.job_properties.get("rule", "rule_name")
+
+    @property
+    def groupid(self):
+        return self.job_properties.get("groupid", "group")
+
+    @property
+    def is_group_jobtype(self):
+        return self.job_properties.get("type", "") == "group"
+
+    @property
+    def jobname(self) -> str:
+        if self.is_group_jobtype:
+            return "{groupid}_{jobid}".format(groupid=self.groupid, jobid=self.jobid)
+        else:
+            return self.cluster.get("jobname",
+                                    "{rule_name}.{wildcards_str}".format(rule_name=self.rule_name,
+                                                                         wildcards_str=self.wildcards_str))
+
+    @property
+    def jobid(self):
+        if self.is_group_jobtype:
+            return self.job_properties.get("jobid", "").split("-")[0]
+        else:
+            return self.job_properties.get("jobid")
+
+    @property
+    def logdir(self):
+        return Path(self.cluster.get("logdir", CookieCutter.get_log_dir()))
+
+    @property
+    def outlog(self):
```

Comment:
I know it's annoying but could you please add type annotations?

Reply:
Done in https://github.com/leoisl/snakemake-lsf/commit/4eb7fe3e2c026e40c9e34d1a510759aca1f040cd